DFC.ServiceTaxonomy.Neo4j/Extensions/DriverExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using Neo4j.Driver;
+using System.Linq;
+using DFC.ServiceTaxonomy.Neo4j.Models.Interface;
+
+namespace DFC.ServiceTaxonomy.Neo4j.Extensions
+{
+    public static class DriverExtensions
+    {
+        public static IDriver PrimaryDriver(this IEnumerable<INeoDriver> drivers)
+        {
+            var primaryDrivers = drivers.Where(x => string.Equals(x.Type, "Primary", StringComparison.CurrentCultureIgnoreCase));
+
+            if(primaryDrivers.Count() > 1)
+            {
+                throw new InvalidOperationException("More than 1 enabled Primary Driver defined in Neo4JConfiguration");
+            }
+
+            if(!primaryDrivers.Any())
+            {
+                throw new InvalidOperationException("No enabled Primary Driver defined in Neo4JConfiguration");
+            }
+
+            return primaryDrivers.FirstOrDefault().Driver;
```

Comment:
Don't think we need `FirstOrDefault()` here, the above code already ensure's there's 1, so `First()` should be fine. (Also means we don't have to worry about the null return.)

Reply:
Driver extensions is no longer relevant since we added round-robin and is being removed